### DISCUSSION TITLE
Generating mipmaps with preprocessing

### DISCUSSION THREAD

**Discussion by dafteran4** [bevyengine/bevy#12336] [Q&A]
Hi, I am trying to figure out how to generate mipmaps for textures. I've read in bevy 0.12 news that one should use `CompressedImageSaver` with asset pre-processing. Unsure of the details, I enabled features `basis-universal` and `asset_processor`, then enabled processing in `AssetPlugin { mode: AssetMode::Processed, ..default() }`. I also load my textures by setting ImageSampleDescriptor to:
```
let settings = move |s: &mut ImageLoaderSettings| {
    let mut desc = ImageSamplerDescriptor::default();
    desc.address_mode_u = ImageAddressMode::Repeat;
    desc.address_mode_v = ImageAddressMode::Repeat;
    desc.mag_filter = ImageFilterMode::Linear;
    desc.min_filter = ImageFilterMode::Linear;
    desc.mipmap_filter = ImageFilterMode::Linear;
    desc.anisotropy_clamp = 16;
    s.sampler = ImageSampler::Descriptor(desc);
};
let tex = server.load_with_settings("texture.png", settings);
```
Yet, I am seeing no mipmaps. I'd appreciate any suggestions on how to make this work. Thanks.

**Comment by Affinator** [comment]
Maybe take a look here: https://github.com/DGriffin91/bevy_mod_mipmap_generator

**Reply by dafteran4** [reply]
Thanks, I did. I ended up manually generating midmaps with the ubasis tool, also linked from that page.
